plugins/modem-manager/fu-mm-device.c

```diff
@@ -189,6 +190,9 @@ fu_mm_device_probe_default (FuDevice *device, GError **error)
 				break;
 			}
 		}
+		/* only set if fastboot wasn't already set */
+		if (fu_device_get_protocol (device) != NULL)
+			fu_device_set_protocol (device, "com.qualcomm.qmi_pdc");
```

Comment:
Is this correct? On a naive reading it seems the code doesn't seem to match the comment.

Reply:
It's due to the order that the function is populated. If fastboot was found higher up it will set it to fastboot. If that wasn't set the protocol should be NULL and if qmi is found it gets set.

Reply:
I assumed fu_device_get_protocol (device) returning NULL when it is not yet set to fastboot, and non-NULL if it was set to fastboot.  The code sets qmi in the latter (non-NULL) case. (i.e. I expected the condition to read fu_device_get_protocol (device) == NULL.)

Reply:
Oh, you're right, great catch! I'll push a follow up fix.